runtime/near-vm-logic/src/gas_counter.rs

```diff
@@ -108,21 +172,36 @@ impl GasCounter {
                 num_bytes.checked_mul(per_byte_fee.exec_fee()).ok_or(HostError::IntegerOverflow)?,
             )
             .ok_or(HostError::IntegerOverflow)?;
-
+        self.update_profile_action(action, burn_gas);
         self.deduct_gas(burn_gas, use_gas)
     }
 
     /// A helper function to pay base cost gas fee for batching an action.
     /// # Args:
     /// * `base_fee`: base fee for the action;
     /// * `sir`: whether the receiver_id is same as the current account ID;
-    pub fn pay_action_base(&mut self, base_fee: &Fee, sir: bool) -> Result<()> {
+    pub fn pay_action_base(&mut self, base_fee: &Fee, sir: bool, action: Actions) -> Result<()> {
```

Comment:
We should just change `pay_action_base` and `pay_acion_per_byte` so that it takes `action` but does not take `base_fee` and then store `RuntimeFeesConfig` in `GasCounter` just like we store `ExtCostsConfig`. Then we would need to implement `value()` method for `Actions` just like we did it here: https://github.com/nearprotocol/nearcore/blob/f024a28d9b0f1b2238f04d288ddebc8ac12ff4d1/runtime/near-vm-logic/src/config.rs#L490

Reply:
This is rather major change, do you believe we shall do that in this (somewhat unrelated) profiler change?

Reply:
Let's create a tracking issue for it and fix it later.